src/components/tables/table-children/TableChildBody.tsx

```diff
@@ -58,25 +58,27 @@ export const TableChildBody = (props: ITableChildBodyProps): JSX.Element => {
         disabled: !!props.rowData.disabled || !_.isUndefined(props.rowData.enabled) && !props.rowData.enabled,
     });
 
-    return (
-        <TableRowWrapper className={tableRowWrapperClasses}>
-            <TableHeadingRowConnected
-                id={headingAndCollapsibleId}
-                tableId={props.tableId}
-                className={tableRowClasses}
-                isCollapsible={!!collapsibleData}
-                onClickCallback={() => {
-                    if (props.onRowClick) {
-                        props.onRowClick(props.getActions(props.rowData));
-                    }
-                }}
-                onDoubleClick={() => props.getActions(props.rowData)
-                    .filter((action) => action.callOnDoubleClick)
-                    .forEach((action) => action.trigger())
-                }>
-                {tableHeadingRowContent}
-            </TableHeadingRowConnected>
-            {collapsibleRow}
-        </TableRowWrapper>
-    );
+    const tableHeadingRowConnectedNode = <TableHeadingRowConnected
+        id={headingAndCollapsibleId}
+        tableId={props.tableId}
+        className={tableRowClasses}
+        isCollapsible={!!collapsibleData}
+        onClickCallback={() => {
+            if (props.onRowClick) {
+                props.onRowClick(props.getActions(props.rowData));
+            }
+        }}
+        onDoubleClick={() => props.getActions(props.rowData)
+            .filter((action) => action.callOnDoubleClick)
+            .forEach((action) => action.trigger())
+        }>
+        {tableHeadingRowContent}
+    </TableHeadingRowConnected>;
+
+    return collapsibleRow
+        ? <TableCollapsibleRowWrapper className={tableRowWrapperClasses}
+            collapsibleRow={collapsibleRow}>
+            {tableHeadingRowConnectedNode}
+        </TableCollapsibleRowWrapper>
+        : tableHeadingRowConnectedNode;

```

Comment:
i think you could just rewrite the end of this Component as such:  
  
```

    const tableHeadingRowConnectedNode = (
        <TableHeadingRowConnected
            id={headingAndCollapsibleId}
            tableId={props.tableId}
            className={tableRowClasses}
            isCollapsible={!!collapsibleData}
            onClickCallback={() => {
                if (props.onRowClick) {
                    props.onRowClick(props.getActions(props.rowData));
                }
            }}
            onDoubleClick={() => props.getActions(props.rowData)
                .filter((action) => action.callOnDoubleClick)
                .forEach((action) => action.trigger())
            }>
            {tableHeadingRowContent}
        </TableHeadingRowConnected>
    );

    return collapsibleRow
        ? (
            <TableRowWrapper className={tableRowWrapperClasses}>
                {tableHeadingRowConnectedNode}
                {collapsibleRow}
            </TableRowWrapper>
        )
        : tableHeadingRowConnectedNode;  
```  
I don't think it is necessary to change the name and structure of TableRowWrapper.tsx imho, and you'll have less UTs to fix :D (you currently have 6)